lte/gateway/c/core/oai/tasks/sgw/spgw_state_manager.cpp

```diff
@@ -58,9 +58,13 @@ void SpgwStateManager::create_state() {
   state_teid_map.set_name(S11_BEARER_CONTEXT_INFO_HT_NAME);
   state_teid_map.bind_callback(spgw_free_s11_bearer_context_information);
 
-  state_ue_ht =
-      hashtable_ts_create(SGW_STATE_CONTEXT_HT_MAX_SIZE, nullptr,
-                          (void (*)(void**))sgw_free_ue_context, nullptr);
+  state_ue_map.map = new google::protobuf::Map<uint64_t, spgw_ue_context_s*>();
+  if (!(state_ue_map.map)) {
```

Comment:
If you use new to instantiate an object in C++ the Null check is always needless because of the bad_alloc exception.